include/lug/Window/Window.hpp

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+//
+// include/lug/Window/Window.hpp
+// bon ici c'est pas trop compliqu�, c'est l'interface pour l'utilisateur, rien de nouveau :)
+// pour g�rer les erreurs a la cr�ation de fenetre je penses qu'il peut �tre int�ressant de passer par une factory qui g�re �a (je le fais pas l�)
+// il faut quand m�me indiquer qu'on a une impl�mentation en membre
+// WARNING: Le code ne respecte pas forc�ment notre guideline, c'est juste un exemple !
+//
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+#include <lug/Window/EventType.hpp>
+#include <lug/Window/Export.hpp>
+#include <queue>
+#include <memory>
+
+namespace lug {
+namespace Window {
+
+		// le nom de ce namespace est le m�me que la SFML, je trouve �a tr�s bien comme nom, je penses que sans l'influence de la SFML je l'aurais appel� "private" donc bon :)
+namespace priv {
+		// on ne sait pas a quoi �a correspond pour le moment, donc on le forward declare
+		// vu qu'on l'utilise que sous forme de pointeur dans la d�claration �a marchera
+		class WindowImpl;
+} // namespace priv
+
+	struct LUG_WINDOW_API Event
+	{
+		EventType type;
+		// a quoi ressemble un event pour l'utilisateur ? :D
+	};
+
+	enum LUG_WINDOW_API Style
+	{
+		None = 0,      ///< No border / title bar (this flag and all others are mutually exclusive)
+		Titlebar = 1 << 0, ///< Title bar + fixed border
+		Resize = 1 << 1, ///< Title bar + resizable border + maximize button
+		Close = 1 << 2, ///< Title bar + close button
+		Fullscreen = 1 << 3, ///< Fullscreen mode (this flag and all others are mutually exclusive)
+
+		Default = Titlebar | Resize | Close ///< Default window style
+	};
```

Comment:
Please align the comments, at least the 5 firsts. And do not try to make them *pretty* (`///<` is just `//`)

Did it for you:
```
        None = 0,               // No border / title bar (this flag and all others are mutually exclusive)
        Titlebar = 1 << 0,      // Title bar + fixed border
        Resize = 1 << 1,        // Title bar + resizable border + maximize button
        Close = 1 << 2,         // Title bar + close button
        Fullscreen = 1 << 3,    // Fullscreen mode (this flag and all others are mutually exclusive)

        Default = Titlebar | Resize | Close     // Default window style
```